backend/controllers/historyController.js

```diff
@@ -7,22 +9,138 @@ const asyncHandler = require('../middleware/async');
+
+    //Vehicle API History
+    else if (req.originalUrl.indexOf('/api/v1/inventory') > - 1) {
+        const vehicle = await Vehicle.findById(req.params.vehicleId);
+        if (req.method == 'DELETE') {
+            body = {
+                vehicle: vehicle._id,
+                author: req.user._id,
+                log: `A vehicle with vin: ${vehicle.vin} has been deleted by ${req.user.first_name} ${req.user.last_name}`,
+            }
+        }
+        else if (req.method == 'PUT') {
+            if (req.body.delivered) {
+                body = {
+                    vehicle: vehicle._id,
+                    author: req.user._id,
+                    log: `A vehicle with vin: ${vehicle.vin} has been marked delivered by ${req.user.first_name} ${req.user.last_name}`,
+                }
+            }
+            else if (vehicle.missing != req.body.missing) {
+                if (req.body.missing) {
+                    body = {
+                        vehicle: vehicle._id,
+                        author: req.user._id,
+                        log: `A vehicle with vin: ${vehicle.vin} has been marked missing by ${req.user.first_name} ${req.user.last_name}`,
+                    }
+                }
+                else {
+                    body = {
+                        vehicle: vehicle._id,
+                        author: req.user._id,
+                        log: `A vehicle with vin: ${vehicle.vin} has been marked found by ${req.user.first_name} ${req.user.last_name}`,
+                    }
+                }
+            }
+        }
+    }
+
+    //Comments API History
+    else if (req.originalUrl.indexOf('/api/v1/comments/') > - 1) {
+        if (req.method=='POST') {
+            const vehicle = await Vehicle.findById(req.params.vehicleId);
+            body = {
+                vehicle: req.params.vehicleId,
+                author: req.user._id,
+                log: `A comment has been added on the vehicle with vin: ${vehicle.vin} by ${req.user.first_name} ${req.user.last_name}`,
+            }
+        }
+        else {
+            const comment = await Comment.findById(req.params.commentId);
+            if (req.method=='DELETE') {
+                body = {
+                    vehicle: comment.vehicle,
+                    author: req.user._id,
+                    log: `A comment has been deleted on the vehicle with vin: ${vehicle.vin} by ${req.user.first_name} ${req.user.last_name}`,
+                }
+            }
+            else if (req.method=='PUT') {
+                body = {
+                    vehicle: comment.vehicle,
+                    author: req.user._id,
+                    log: `A comment has been edited on the vehicle with vin: ${vehicle.vin} by ${req.user.first_name} ${req.user.last_name}`,
+                }
+            }
+        }
+    }
+
+    //if body is defined, then create the event
+    if (Object.keys(body).length !== 0) {
+        const history = await History.create(body);
```

Comment:
Similarly to the previous comment, the history variable is never used so it can be removed, but keeping the function call.

Reply:
This is addressed above.